WallEView/ViewController.swift

```diff
@@ -39,24 +39,34 @@ class ViewController: NSViewController, NSTableViewDataSource, NSTableViewDelega
         didSet {
             tableView.reloadData()
 
-            let hideContent = state?.isIdle == true || state?.isFailing == true
+            // TODO: this logic should pick the queue depending on user selection of target branch
+            let queue = state?.queues.first
+
+            let hideContent = queue?.isIdle == true || queue?.isFailing == true
             background.isHidden = !hideContent
             tableViewContainer.isHidden = hideContent
 
-            if state?.isFailing == true {
+            if state?.queues.first?.isFailing == true {
                 backgroundImage.image = #imageLiteral(resourceName: "foot")
-                backgroundLabel.stringValue = "Something failin':\n\n\(state?.error.map(String.init(describing:)) ?? "")"
+                backgroundLabel.stringValue = "Something failin':\n\n\(queue?.error.map(String.init(describing:)) ?? "")"
             } else {
                 backgroundImage.image = #imageLiteral(resourceName: "green")
                 backgroundLabel.stringValue = "Doin' nothin', just chillin'"
             }
         }
     }
 
-    struct State {
+    struct State: Decodable {
+        let queues: [Queue]
+
+        static let empty = State(queues: [])
+    }
+
+    struct Queue {
         struct Current: Decodable {
             let reference: PullRequest
         }
+        let targetBranch: String
         let current: Current?
         let queue: [PullRequest]
```

Comment:
[nit] rename `queue` to `pending` maybe? Since it feels strange and not easy to understand having a `Queue` type that _contains_ a `queue`… 🤔 (not that important though)